src/shared/lib/MutationAnnotator.ts

```diff
@@ -99,6 +100,68 @@ export function filterMutationByTranscriptId(mutation:Mutation,
     }
 }
 
+export function getMutationToTranscriptId(mutation:Mutation,
+                                          ensemblTranscriptId: string,
+                                          indexedVariantAnnotations: {[genomicLocation: string]: VariantAnnotation})
+{
+    const genomicLocation = extractGenomicLocation(mutation);
+    const variantAnnotation = genomicLocation ?
+        indexedVariantAnnotations[genomicLocationString(genomicLocation)] : undefined;
+    
+    const transcriptConsequenceSummaries = variantAnnotation && 
+        variantAnnotation.annotation_summary &&  
+        variantAnnotation.annotation_summary.transcriptConsequenceSummaries &&  
+        variantAnnotation.annotation_summary.transcriptConsequenceSummaries.filter((tc:TranscriptConsequenceSummary) => tc.transcriptId === ensemblTranscriptId);
+
+    if (variantAnnotation && transcriptConsequenceSummaries && transcriptConsequenceSummaries.length > 0) {
+        const transcriptConsequenceSummary = transcriptConsequenceSummaries[0]; // TODO: should pick most impactful one
+        return getMutationFromSummary(mutation, variantAnnotation.annotation_summary, transcriptConsequenceSummary, true);
+    } else {
+        return undefined;
+    }
+}
+
+export function getMutationFromSummary(mutation:Mutation,
+                                       annotationSummary:VariantAnnotationSummary,
+                                       transcriptConsequenceSummary:TranscriptConsequenceSummary,
+                                       overwrite:boolean)
```

Comment:
maybe add a default value `false` for overwrite?

Reply:
i think it's good to be explicit in this case, it's called once with false, once with true, so there's not really a clear preference over which should be the default.

`overwrite` parameter might be slightly confusing, cause initAnnotatedMutation always deepclones the entire mutation. It's either using the existing values or overwriting them in the new object